File: backend/internal/model/session.go
package model

import "time"

// Session 上机记录。
type Session struct {
	ID              uint       `gorm:"primaryKey" json:"id"`
	UserID          uint       `gorm:"index;not null" json:"user_id"`
	StationID       uint       `gorm:"index;not null" json:"station_id"`
	ReservationID   uint       `gorm:"index" json:"reservation_id"`
	StartTime       time.Time  `gorm:"not null" json:"start_time"`
	EndTime         *time.Time `json:"end_time"`
	DurationMinutes int        `gorm:"default:0" json:"duration_minutes"`
	GameType        string     `gorm:"size:16;default:other" json:"game_type"`
	Amount          float64    `gorm:"type:decimal(12,2);default:0" json:"amount"`
	Status          string     `gorm:"size:16;default:active" json:"status"` // active/completed
	CreatedAt       time.Time  `json:"created_at"`
	UpdatedAt       time.Time  `json:"updated_at"`
}

// TableName 指定表名。
func (Session) TableName() string { return "sessions" }

File: frontend/src/api/session.ts
import { get, post } from '@/utils/request'

export interface Session {
  id: number
  user_id: number
  station_id: number
  start_time: string
  end_time: string | null
  duration_minutes: number
  game_type: string
  amount: number
  status: string
}

export interface RankItem {
  rank: number
  user_id: number
  username: string
  nickname: string
  total_minutes: number
}

export function listSessions(params: { page: number; page_size: number; status?: string; user_id?: number }) {
  return get<{ list: Session[]; total: number }>('/sessions', params)
}

export function startSession(data: { station_id: number; reservation_id?: number; game_type?: string }) {
  return post<Session>('/sessions', data)
}

export function renewSession(id: number, add_minutes: number) {
  return post<Session>(`/sessions/${id}/renew`, { add_minutes })
}

export function endSession(id: number, game_type?: string) {
  return post<Session>(`/sessions/${id}/end`, { game_type })
}

export function getRank(params: { period?: string; game_type?: string; limit?: number }) {
  return get<RankItem[]>('/sessions/rank', params)
}
